gensrc/thrift/Types.thrift

```diff
@@ -126,6 +126,9 @@ struct TTypeNode {
 
     // only used for structs; has struct_fields.size() corresponding child types
     3: optional list<TStructField> struct_fields
+
+    // only used for complex types, such as array, map and etc.
+    4: optional bool contains_null
```

Comment:
use is_null or is_nullable instead of contains_null;

is_null means current scalar_type is null or not.